app/core/core/src/main/java/io/syndesis/core/KeyGenerator.java

```diff
@@ -59,7 +59,7 @@ public static String createKey() {
         buffer.putLong(getRandomPart(now));
 
         try {
-            return Base64.encodeBytes(buffer.array(), 2, 15, Base64.ORDERED);
+            return Names.sanitize(Base64.encodeBytes(buffer.array(), 2, 15, Base64.ORDERED));
```

Comment:
This breaks the KeyGeneratorTest.  Generated keys are supposed to be lexically incrementing always.

Reply:
I think the only thing you need to do to keys to make them valid labels in k8s is prefix them with a letter.